Use before definition errors should not be reported in ambient contexts

```ts
// test.d.ts

declare var S: typeof A; // error TS2448: Block-scoped variable 'A' used before its declaration.
declare const A: number;  
```

Errors make sense only in non-ambient context.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -976,7 +976,7 @@ namespace ts {
 
             Debug.assert(declaration !== undefined, "Block-scoped variable declaration is undefined");
 
-            if (!isBlockScopedNameDeclaredBeforeUse(<Declaration>getAncestor(declaration, SyntaxKind.VariableDeclaration), errorLocation)) {
+            if (!isInAmbientContext(declaration) && !isBlockScopedNameDeclaredBeforeUse(<Declaration>getAncestor(declaration, SyntaxKind.VariableDeclaration), errorLocation)) {
                 error(errorLocation, Diagnostics.Block_scoped_variable_0_used_before_its_declaration, declarationNameToString(declaration.name));
             }
         }

--- a/tests/baselines/reference/noUsedBeforeDefinedErrorInAmbientContext1.symbols
+++ b/tests/baselines/reference/noUsedBeforeDefinedErrorInAmbientContext1.symbols
@@ -1,0 +1,9 @@
+=== tests/cases/compiler/test.d.ts ===
+
+declare var S: typeof A; // no error
+>S : Symbol(S, Decl(test.d.ts, 1, 11))
+>A : Symbol(A, Decl(test.d.ts, 2, 13))
+
+declare const A: number;  
+>A : Symbol(A, Decl(test.d.ts, 2, 13))
+

--- a/tests/baselines/reference/noUsedBeforeDefinedErrorInAmbientContext1.types
+++ b/tests/baselines/reference/noUsedBeforeDefinedErrorInAmbientContext1.types
@@ -1,0 +1,9 @@
+=== tests/cases/compiler/test.d.ts ===
+
+declare var S: typeof A; // no error
+>S : number
+>A : number
+
+declare const A: number;  
+>A : number
+

--- a/tests/cases/compiler/noUsedBeforeDefinedErrorInAmbientContext1.ts
+++ b/tests/cases/compiler/noUsedBeforeDefinedErrorInAmbientContext1.ts
@@ -1,0 +1,4 @@
+﻿// @filename: test.d.ts
+
+declare var S: typeof A; // no error
+declare const A: number;  

